src/input/pointer/mod.rs

```diff
@@ -714,44 +714,43 @@ impl<D: SeatHandler + 'static> PointerInternal<D> {
         &mut self,
         data: &mut D,
         seat: &Seat<D>,
-        focus: Option<(<D as SeatHandler>::PointerFocus, Point<i32, Logical>)>,
+        focus: Option<RelativePoint<D::PointerFocus>>,
         event: &MotionEvent,
     ) {
         // do we leave a surface ?
         let mut leave = true;
         self.location = event.location;
-        if let Some((ref current_focus, _)) = self.focus {
-            if let Some((ref new_focus, _)) = focus {
-                if current_focus == new_focus {
+        if let Some(ref current_focus) = self.focus {
+            if let Some(ref new_focus) = focus {
+                if current_focus.surface == new_focus.surface {
                     leave = false;
                 }
             }
         }
         if leave {
-            if let Some((focused, _)) = self.focus.as_mut() {
-                focused.leave(seat, data, event.serial, event.time);
+            if let Some(focus) = self.focus.as_mut() {
+                focus.surface.leave(seat, data, event.serial, event.time);
             }
             self.focus = None;
             data.cursor_image(seat, CursorImageStatus::default_named());
         }
 
         // do we enter one ?
-        if let Some((surface, surface_location)) = focus {
+        if let Some(focus) = focus {
             let entered = self.focus.is_none();
             // in all cases, update the focus, the coordinates of the surface
             // might have changed
-            self.focus = Some((surface, surface_location));
+            self.focus = Some(focus.clone());
             let event = MotionEvent {
-                location: event.location - surface_location.to_f64(),
+                location: focus.loc.to_f64(),
```

Comment:
So we are not using the `MotionEvent` location anymore at all? I quite dislike that to be honest, because that means we pass in multiple locations, but ignore some of them. In that case please change the signature of `motion` to **not** take a `MotionEvent`, but directly the `serial` and `time`.

Reply:
`event.location` is still used to set `self.location`, which is provided by `PointerHandle.current_location`, for example.

Reply:
> event.location is still used to set self.location

Right... Maybe we should then at least rename `MotionEvent::location` to `global_location` or `absolute_location` to further differenciate the two values?

Reply:
Done in e3a992725396e8aefb088d9836c962e6512375a3.